### PR TITLE
Validate object identifiers 

### LINKED ISSUE #1007: Define and enforce naming convention for index ID, source ID, ...
`[a-zA-Z0-9_-]{1,255}` ?

Following https://github.com/quickwit-oss/quickwit/pull/1272/files we need to make sure that source_id do not contain any of the kafka group_id unallowed characters, or fix the kafka source.

### PR DESCRIPTION
**Pull request by guilload**
### Description
Validate index ID, source ID, ... Fixes #1007.

### How was this PR tested?
- added unit test
- ran `make test-all`

### CODE REVIEW

File: quickwit-core/src/index.rs [hunk 1]
```diff
@@ -91,6 +93,9 @@ impl IndexService {
         &self,
         index_config: IndexConfig,
     ) -> Result<IndexMetadata, IndexServiceError> {
+        index_config
```
Comment (guilload) [inline quickwit-core/src/index.rs:96]: Injecting the validation logic in `IndexService`. Ideally, we would validate identifiers in the metastore trait before creating objects.

**Comment by guilload** [comment]
PTAL

File: quickwit-config/src/lib.rs [hunk 2]
```diff
@@ -33,3 +37,36 @@ pub use source_config::{
     FileSourceParams, KafkaSourceParams, KinesisSourceParams, PushApiSourceParams,
     RegionOrEndpoint, SourceConfig, SourceParams, VecSourceParams, VoidSourceParams,
 };
+
+fn validate_identifier(label: &str, value: &str) -> anyhow::Result<()> {
+    static IDENTIFIER_REGEX: OnceCell<Regex> = OnceCell::new();
+
+    if IDENTIFIER_REGEX
+        .get_or_init(|| Regex::new(r"^[a-zA-Z0-9][a-zA-Z0-9-_]{2,254}$").expect("Failed to compile regular expression. This should never happen! Please, report on https://github.com/quickwit-oss/quickwit/issues."))
```
[the remark this answers is not in the document]
Comment (fulmicoton) [reply quickwit-config/src/lib.rs:46]: Should we replace a-zA-Z and use \w?

**Review by fulmicoton — APPROVED** [review]

File: quickwit-config/src/lib.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (fmassot) [reply quickwit-config/src/lib.rs:46]: I think that "." is common in names and I kind of like using it.

AFAIK we want `\w` but not for the first character? The idea is to reserve that for internal usage?

**Pull request merged**